doc/source/cluster/kubernetes/examples/rayjob-kueue-priority-scheduling.md

```diff
@@ -0,0 +1,317 @@
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: ResourceFlavor
+metadata:
+  name: "default-flavor"
+---
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: ClusterQueue
+metadata:
+  name: "cluster-queue"
+spec:
+  preemption:
+    reclaimWithinCohort: Any
+    withinClusterQueue: LowerPriority
+  namespaceSelector: {} # match all.
+  resourceGroups:
+  - coveredResources: ["cpu", "memory", "nvidia.com/gpu"]
+    flavors:
+    - name: "default-flavor"
+      resources:
+      - name: "cpu"
+        nominalQuota: 2
+      - name: "memory"
+        nominalQuota: 8G
+      - name: "nvidia.com/gpu"
+        nominalQuota: 1
+---
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: LocalQueue
+metadata:
+  namespace: "default"
+  name: "user-queue"
+spec:
+  clusterQueue: "cluster-queue"
+---
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: WorkloadPriorityClass
+metadata:
+  name: prod-priority
+value: 1000
+description: "Priority class for prod jobs"
+---
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: WorkloadPriorityClass
+metadata:
+  name: dev-priority
+value: 100
+description: "Priority class for development jobs"
+```
+
+Create the Kueue resources:
+```bash
+kubectl apply -f kueue-resources.yaml
+```
+
+## Deploy a RayJob
+
+Download the RayJob that that will execute all the steps documented in [Fine-tune a PyTorch Lightning Text Classifier](https://docs.ray.io/en/master/train/examples/lightning/lightning_cola_advanced.html). The [source code](https://github.com/ray-project/kuberay/tree/master/ray-operator/config/samples/pytorch-text-classifier) is also in the kuberay repo.
+
+```bash
+curl -LO https://raw.githubusercontent.com/ray-project/kuberay/master/ray-operator/config/samples/pytorch-text-classifier/ray-job.pytorch-distributed-training.yaml
+```
+
+Before creating the RayJob, modify the RayJob metadata with:
+* a label to assign the RayJob to the LocalQueue created earlier.
+* a label to assign the RayJob with the `dev-priority` priority class.
+* Modify the name to indicate this is a job for development purposes.
+
+```yaml
+metadata:
+  generateName: dev-pytorch-text-classifier-
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/priority-class: dev-priority
+```
+
+Also note the resources required for this RayJob by looking at the resources requested by the nodes:
+```yaml
+resources:
+  limits:
+    cpu: "8"
+    memory: "20G"
+    nvidia.com/gpu: "1"
+  requests:
+    cpu: "8"
+    memory: "20G"
+    nvidia.com/gpu: "1"
+```
+
+Now deploy the RayJob:
+```bash
+$ kubectl create -f ray-job.pytorch-distributed-training.yaml
+rayjob.ray.io/dev-pytorch-text-classifier-r6d4p created
+```
+
+Verify that the RayCluster and Ray job are running:
+```bash
+$ kubectl get pod
+NAME                                                      READY   STATUS    RESTARTS   AGE
+dev-pytorch-text-classifier-r6d4p-4nczg                   1/1     Running   0          4s
+torch-text-classifier-r6d4p-raycluster-br45j-head-8bbwt   1/1     Running   0          34s
+```
+
```

Comment:
Delete the RayJob after the RayJob completes to make the result of each step in the next section "Queuing Multiple RayJob custom resources" more predictable.